### PR TITLE
Fix #11714: AJAX update containing resources check for duplicates

### LINKED ISSUE #11714: Core: AJAX update of resource doesn't check for duplicate CSS/JS
### Describe the bug

@tandraschko this was originally reported on a forum post in 2022: https://forum.primefaces.org/viewtopic.php?t=71210

We never got to the bottom of what was happening then...but my current client has this same issue.  They do a LOT of dynamic removing and adding components to a single page app using `PrimeFaces.current().ajax().update()` as well as dynamically constructing components in Java.   its a HUGE project.

Here you can see duplicates in the head like `InputNumber` and `FileUpload`.  If I close one of their panels and open it again it will add 2 more copies. 

I shortened it for brevity but in `javax.faces.Resource` update the scripts are there again.

```xml
<?xml version='1.0' encoding='UTF-8'?>
<partial-response>
	<changes>
		<update id="javax.faces.Resource"><![CDATA[
<link type="text/css" rel="stylesheet" href="/emro/javax.faces.resource/fileupload/fileupload.css.trax?ln=primefaces&amp;v=10.0.20&amp;e=10.0.7" />
<script type="text/javascript" src="/emro/javax.faces.resource/fileupload/fileupload.js.trax?ln=primefaces&amp;v=10.0.20&amp;e=10.0.7"></script>
<script type="text/javascript" src="/emro/javax.faces.resource/inputnumber/inputnumber.js.trax?ln=primefaces&amp;v=10.0.20&amp;e=10.0.7"></script>
<link type="text/css" rel="stylesheet" href="/emro/javax.faces.resource/inputnumber/inputnumber.css.trax?ln=primefaces&amp;v=10.0.20&amp;e=10.0.7" />
<script type="text/javascript" src="/emro/javax.faces.resource/inputmask/inputmask.js.trax?ln=primefaces&amp;v=10.0.20&amp;e=10.0.7"></script>]]></update>
		<update id="mainForm:mainPnl"><![CDATA[<div id="mainForm:mainPnl" </div>]]></update>
		<update id="j_id1:javax.faces.ViewState:0"><![CDATA[-1272965173743229381:1530783905304280517]]></update>
	</changes>
</partial-response>
```

Our `core.ajax` for those just automatically appends to the `head` without checking for existing CSS or JS already loaded.

```js
else if (id === PrimeFaces.ajax.RESOURCE) {
    $('head').append(content);
}
```

And thus over time our head is filling with more and more scripts and is part of their overall browser slowdown after using the app for an hour...

```xml
<head id="j_idt2">
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="apple-mobile-web-app-capable" content="yes">
<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0">
<link href="/emro/resources/css/fonts.css" rel="stylesheet" type="text/css">
<link href="/emro/resources/css/icons.css" rel="stylesheet" type="text/css">
<link rel="apple-touch-icon" href="/emro/resources/bmp/eMRO_WebIcon.png">
<link rel="shortcut icon" type="image/x-icon" href="/emro/resources/bmp/eMRO_Web.ico">
<link type="text/css" rel="stylesheet" href="/emro/javax.faces.resource/blockui/blockui.css.trax?ln=primefaces-extensions&amp;v=10.0.20&amp;e=10.0.7">
<link type="text/css" rel="stylesheet" href="/emro/javax.faces.resource/bootstrap.css.trax?ln=css">
<link type="text/css" rel="stylesheet" href="/emro/javax.faces.resource/components.css.trax?ln=primefaces&amp;v=10.0.20&amp;e=10.0.7">
<link type="text/css" rel="stylesheet" href="/emro/javax.faces.resource/css/layout-blue.css.trax?ln=california-layout">
<link type="text/css" rel="stylesheet" href="/emro/javax.faces.resource/css/nanoscroller.css.trax?ln=california-layout">
<link type="text/css" rel="stylesheet" href="/emro/javax.faces.resource/fa/font-awesome.css.trax?ln=primefaces&amp;v=10.0.20&amp;e=10.0.7">
<link type="text/css" rel="stylesheet" href="/emro/javax.faces.resource/fileupload/fileupload.css.trax?ln=primefaces&amp;v=10.0.20&amp;e=10.0.7">
<link type="text/css" rel="stylesheet" href="/emro/javax.faces.resource/fileupload/fileupload.css.trax?ln=primefaces&amp;v=10.0.20&amp;e=10.0.7">
<link type="text/css" rel="stylesheet" href="/emro/javax.faces.resource/inputnumber/inputnumber.css.trax?ln=primefaces&amp;v=10.0.20&amp;e=10.0.7">
<link type="text/css" rel="stylesheet" href="/emro/javax.faces.resource/inputnumber/inputnumber.css.trax?ln=primefaces&amp;v=10.0.20&amp;e=10.0.7">
<link type="text/css" rel="stylesheet" href="/emro/javax.faces.resource/inputnumber/inputnumber.css.trax?ln=primefaces&amp;v=10.0.20&amp;e=10.0.7">
<link type="text/css" rel="stylesheet" href="/emro/javax.faces.resource/local.css.trax?ln=css">
<link type="text/css" rel="stylesheet" href="/emro/javax.faces.resource/primeicons/primeicons.css.trax?ln=primefaces&amp;v=10.0.20&amp;e=10.0.7">
<link type="text/css" rel="stylesheet" href="/emro/javax.faces.resource/theme.css.trax?ln=primefaces-california-blue&amp;v=10.0.20&amp;e=10.0.7">

<script type="text/javascript" src="/emro/javax.faces.resource/blockui/blockui.js.trax?ln=primefaces-extensions&amp;v=10.0.20&amp;e=10.0.7"></script>
<script type="text/javascript" src="/emro/javax.faces.resource/chartjs/chartjs.js.trax?ln=primefaces&amp;v=10.0.20&amp;e=10.0.7"></script>
<script type="text/javascript" src="/emro/javax.faces.resource/components.js.trax?ln=primefaces&amp;v=10.0.20&amp;e=10.0.7"></script>
<script type="text/javascript" src="/emro/javax.faces.resource/core.js.trax?ln=primefaces&amp;v=10.0.20&amp;e=10.0.7"></script>
<script type="text/javascript" src="/emro/javax.faces.resource/filedownload/filedownload.js.trax?ln=primefaces&amp;v=10.0.20&amp;e=10.0.7"></script>
<script type="text/javascript" src="/emro/javax.faces.resource/fileupload/fileupload.js.trax?ln=primefaces&amp;v=10.0.20&amp;e=10.0.7"></script>
<script type="text/javascript" src="/emro/javax.faces.resource/fileupload/fileupload.js.trax?ln=primefaces&amp;v=10.0.20&amp;e=10.0.7"></script>
<script type="text/javascript" src="/emro/javax.faces.resource/inputmask/inputmask.js.trax?ln=primefaces&amp;v=10.0.20&amp;e=10.0.7"></script>
<script type="text/javascript" src="/emro/javax.faces.resource/inputmask/inputmask.js.trax?ln=primefaces&amp;v=10.0.20&amp;e=10.0.7"></script>
<script type="text/javascript" src="/emro/javax.faces.resource/inputmask/inputmask.js.trax?ln=primefaces&amp;v=10.0.20&amp;e=10.0.7"></script>
<script type="text/javascript" src="/emro/javax.faces.resource/inputnumber/inputnumber.js.trax?ln=primefaces&amp;v=10.0.20&amp;e=10.0.7"></script>
<script type="text/javascript" src="/emro/javax.faces.resource/inputnumber/inputnumber.js.trax?ln=primefaces&amp;v=10.0.20&amp;e=10.0.7"></script>
<script type="text/javascript" src="/emro/javax.faces.resource/inputnumber/inputnumber.js.trax?ln=primefaces&amp;v=10.0.20&amp;e=10.0.7"></script>
<script type="text/javascript" src="/emro/javax.faces.resource/jquery/jquery-plugins.js.trax?ln=primefaces&amp;v=10.0.20&amp;e=10.0.7"></script>
<script type="text/javascript" src="/emro/javax.faces.resource/jquery/jquery.js.trax?ln=primefaces&amp;v=10.0.20&amp;e=10.0.7"></script>
<script type="text/javascript" src="/emro/javax.faces.resource/js/layout.js.trax?ln=california-layout"></script>
<script type="text/javascript" src="/emro/javax.faces.resource/js/nanoscroller.js.trax?ln=california-layout"></script>
<script type="text/javascript" src="/emro/javax.faces.resource/moment/moment.js.trax?ln=primefaces&amp;v=10.0.20&amp;e=10.0.7"></script>
<script type="text/javascript" src="/emro/javax.faces.resource/primefaces-extensions.js.trax?ln=primefaces-extensions&amp;v=10.0.20&amp;e=10.0.7"></script>
<script type="text/javascript">if(window.PrimeFaces){PrimeFaces.settings.locale='en_US';PrimeFaces.settings.viewId='/Home.xhtml';PrimeFaces.settings.contextPath='/emro';PrimeFaces.settings.cookiesSecure=true;}</script>
</head>
```

### Reproducer

I have no idea quite how to reproduce it but I am seeing it in their code base same as that user from 2022.  They are not using a Custom HeadRenderer or anything.

### Expected behavior

No duplicate CSS or JS even if the head is sent duplicate scripts.

### PrimeFaces edition

Community

### PrimeFaces version

13.0.0

### Theme

California

### JSF implementation

Mojarra

### JSF version

2.3

### Java version

11

### Browser(s)

_No response_

### PR DESCRIPTION
**Pull request by melloware**
Fix #11714: AJAX update containing resources check for duplicates

### CODE REVIEW

**Comment by melloware** [comment]
as a note I have tested this heavily with my client project and verified it will only ever add new resources to the Head.

File: primefaces/src/main/resources/META-INF/resources/primefaces/core/core.ajax.js [hunk 1]
```diff
@@ -315,7 +315,21 @@ if (!PrimeFaces.ajax) {
...
+                    // checking if resource is already attached to the head and adding it if not
+                    $(content)
+                        .filter("link[href], script[src]")
+                        .each(function() {
+                            var $resource = $(this);
+                            var src = $resource.attr("href") || $resource.attr("src");
+                            var type = this.tagName.toLowerCase();
+
+                            // Check if script or stylesheet already exists and add it to head if it does not
+                            if (!$("head").find(type + '[src="' + src + '"], ' + type + '[href="' + src + '"]').length) {
```
Comment (tandraschko) [inline primefaces/src/main/resources/META-INF/resources/primefaces/core/core.ajax.js]: just cosmetics as bit more readable:
i would add this to a var `$("head").find(type + '[src="' + src + '"], ' + type + '[href="' + src + '"]')`
and better write:
if ($resources.length == 0)

Comment (melloware) [reply primefaces/src/main/resources/META-INF/resources/primefaces/core/core.ajax.js]: Good feedback.  Pushed.

File: primefaces/src/main/resources/META-INF/resources/primefaces/core/core.ajax.js [hunk 2]
```diff
@@ -315,7 +315,22 @@ if (!PrimeFaces.ajax) {
...
+                        .each(function() {
+                            var $resource = $(this);
+                            var src = $resource.attr("href") || $resource.attr("src");
+                            var type = this.tagName.toLowerCase();
+                            var $resources = $("head").find(type + '[src="' + src + '"], ' + type + '[href="' + src + '"]');
+
+                            // Check if script or stylesheet already exists and add it to head if it does not
+                            if ($resources.length === 0) {
+                                PrimeFaces.debug("Appending " + type + " to head: " + src);
+                                $("head").append($resource);
```
Comment (tandraschko) [inline primefaces/src/main/resources/META-INF/resources/primefaces/core/core.ajax.js]: you may also store head in a var

Comment (melloware) [reply primefaces/src/main/resources/META-INF/resources/primefaces/core/core.ajax.js]: Fixed

**Pull request merged**